core/templates/pages/exploration-player-page/learner-experience/conversation-skin.directive.ts

```diff
@@ -506,7 +506,7 @@ angular.module('oppia').directive('conversationSkin', [
                 // Sometimes setting iframe height to the exact content height
                 // still produces scrollbar, so adding 50 extra px.
                 newHeight += 50;
-                MessengerService.sendMessage(MessengerService.HEIGHT_CHANGE, {
+                MessengerService.sendMessage('HEIGHT_CHANGE', {
```

Comment:
Have changed the implementation slightly [here]( https://github.com/oppia/oppia/pull/13368/files#diff-c22f802d64e2299c875d07099008824015bfd25b1a449b7247d80825a341588eR158)

```typescript
type MessengerPayloadKey = keyof typeof ServicesConstants.MESSENGER_PAYLOAD;
...
sendMessage(messageTitle: MessengerPayloadKey, messageData: Object): void { 
  ...
}
```

I felt like this implementation is more strict whenever we use `sendMessage()`. we cannot call it with another messageTitle except the accepted ones which are the keys in `ServicesConstants.MESSENGER_PAYLOAD`. 

if we keep the type as `string`, 
```typescript
sendMessage(messageTitle: string, messageData: Object): void { ...
```
then we would need to cater the situation when an invalid messageTitle is passed as an argument. This could be done but the above approach seemed more clean to me.

Reply:
Hmm, I see, but can we still use the constants here? It should be possible to type the constants correctly, right?

Reply:
Have changed the implementation again to use constants instead.